files/default/test-neutron-ha-tool.py

```diff
@@ -0,0 +1,168 @@
+import unittest
+import collections
+ha_tool = __import__("neutron-ha-tool")
+
+
+class MockNeutronClient(object):
+
+    def __init__(self):
+        self.routers = {}
+        self.agents = {}
+        self.routers_by_agent = collections.defaultdict(set)
+
+    def tst_add_agent(self, agent_id, props):
+        self.agents[agent_id] = dict(props, id=agent_id)
+
+    def tst_add_router(self, agent_id, router_id, props):
+        self.routers[router_id] = dict(props, id=router_id)
+        self.routers_by_agent[agent_id].add(router_id)
+
+    def tst_agent_by_router(self, router_id):
+        for agent_id, router_ids in self.routers_by_agent.items():
+            if router_id in router_ids:
+                return self.agents[agent_id]
+
+        raise NotImplementedError()
+
+    def list_agents(self):
+        return {
+            'agents': self.agents.values()
+        }
+
+    def list_routers_on_l3_agent(self, agent_id):
+        return {
+            'routers': [
+                self.routers[router_id]
+                for router_id in self.routers_by_agent[agent_id]
+            ]
+        }
+
+    def remove_router_from_l3_agent(self, agent_id, router_id):
+        self.routers_by_agent[agent_id].remove(router_id)
+
+    def add_router_to_l3_agent(self, agent_id, router_body):
+        self.routers_by_agent[agent_id].add(router_body['router_id'])
+
+    def list_ports(self, device_id, fields):
+        return {
+            'ports': [
+                {
+                    'id': 'someid',
+                    'binding:host_id':
+                        self.tst_agent_by_router(device_id)['host'],
+                    'binding:vif_type': 'non distributed',
+                    'status': 'ACTIVE'
+                }
+            ]
+        }
+
+    def list_floatingips(self, router_id):
+        return {
+            'floatingips': [
+                {
+                    'id': 'irrelevant',
+                    'status': 'ACTIVE'
+                }
+            ]
+        }
+
+
+class TestL3AgentMigrate(unittest.TestCase):
+
+    def setUp(self):
+        import logging
+        logging.basicConfig(level=logging.DEBUG)
+
+    def test_no_agents_returns_zero(self):
+        neutron_client = MockNeutronClient()
+        result = ha_tool.l3_agent_migrate(neutron_client)
+
+        self.assertEqual(0, result)
```

Comment:
Doesn't this assume that `MockNeutronClient` starts with no agents?  That sounds a bit brittle - better to assert that as part of the test.